blocks/how-to/src/app.tsx

```diff
@@ -0,0 +1,449 @@
+
+export const App: BlockComponent<RootEntity> = ({
+  graph: { blockEntitySubgraph, readonly },
+}) => {
+  if (!blockEntitySubgraph) {
+    throw new Error("No blockEntitySubgraph provided");
+  }
+  const blockRootRef = useRef<HTMLDivElement>(null);
+  const { graphModule } = useGraphBlockModule(blockRootRef);
+  const { rootEntity: blockEntity, linkedEntities } =
+    useEntitySubgraph(blockEntitySubgraph);
+
+  const {
+    metadata: {
+      recordId: { entityId },
+      entityTypeId,
+    },
+    properties,
+  } = blockEntity;
+
+  const { [titleKey]: title, [descriptionKey]: description } = properties;
+
+  const introLinkedEntity: LinkEntityAndRightEntity = useMemo(
+    () =>
+      linkedEntities.find(
+        ({ linkEntity }) =>
+          linkEntity?.metadata.entityTypeId === introductionLinkType,
+      ),
+    [linkedEntities],
+  )!;
+
+  const introEntity: HowToStep | undefined = introLinkedEntity?.rightEntity;
+  const introLinkEntity: IntroductionLink | undefined =
+    introLinkedEntity?.linkEntity;
+
+  const stepLinkedEntities: LinkEntityAndRightEntity[] = useMemo(
+    () =>
+      linkedEntities.filter(
+        ({ linkEntity }) => linkEntity?.metadata.entityTypeId === stepLinkType,
+      ),
+    [linkedEntities],
+  )!;
+
+  const stepEntities: HowToStep[] | undefined = stepLinkedEntities?.map(
+    (linkEntity) => linkEntity.rightEntity,
+  );
+
+  const [hovered, setHovered] = useState(false);
+  const [titleValue, setTitleValue] = useState(title);
+  const [descriptionValue, setDescriptionValue] = useState(description);
+  const [introAnimatingIn, setIntroAnimatingIn] = useState(false);
+  const [introAnimatingOut, setIntroAnimatingOut] = useState(false);
+  const [stepAnimatingOut, setStepAnimatingOut] = useState(-1);
+
+  const updateField = async (value: string, field: TitleOrDescription) => {
+    await graphModule?.updateEntity({
+      data: {
+        entityId,
+        entityTypeId,
+        properties: {
+          ...properties,
+          [field]: value,
+        },
+      },
+    });
+  };
+
+  const createStepEntity = useCallback(
+    async (linkType: Link) => {
+      if (readonly) {
+        return;
+      }
+
+      const createEntityResponse = await graphModule?.createEntity({
+        data: {
+          entityTypeId: howToStepType,
+          properties: {},
+        },
+      });
+
+      const createdEntityId =
+        createEntityResponse?.data?.metadata.recordId.entityId;
+
+      if (createdEntityId) {
+        await graphModule?.createEntity({
+          data: {
+            entityTypeId: linkType,
+            properties: {},
+            linkData: {
+              leftEntityId: entityId,
+              rightEntityId: createdEntityId,
+            },
+          },
+        });
+      }
+    },
+    [graphModule],
+  );
+
+  const createIntroduction = async () => {
+    setIntroAnimatingIn(true);
+    await createStepEntity(introductionLinkType);
```

Comment:
Haven't really tested outside MBD, since HASH's pages are broken I'm not sure how I could without fixing that-

Disabled the button while the request is happening in ce27afa0832642c52e82dca82a0c7b6c41baa7d0.